### PR TITLE
Fix the traversal of error messages to match the new interface

### PR DESCRIPTION
**Pull request by dkubb**
The error key is no longer a Hash, but is an Array of Hashes. This code traverses them and joins together the descriptions into a single message for the shipping error object.

### CODE REVIEW

File: app/models/spree/calculator/shipping/active_shipping/base.rb [hunk 1]
```diff
@@ -123,16 +123,10 @@ def retrieve_rates(origin, destination, shipment_packages)
...
-              if params.has_key?("Response") && params["Response"].has_key?("Error") && params["Response"]["Error"].has_key?("ErrorDescription")
-                message = params["Response"]["Error"]["ErrorDescription"]
-              else
-                message = e.message
-              end
-            else
-              message = e.message
-            end
+            message = if [ActiveMerchant::ResponseError, ActiveMerchant::Shipping::ResponseError].include?(e.class) && e.response.is_a?(ActiveMerchant::Shipping::Response)
+              errors  = e.response.params.dig('Response', 'Error')
```
Comment (mbj) [inline app/models/spree/calculator/shipping/active_shipping/base.rb]: There is an extra whitespace around the assignment operator.

File: app/models/spree/calculator/shipping/active_shipping/base.rb [hunk 2]
```diff
@@ -123,16 +123,10 @@ def retrieve_rates(origin, destination, shipment_packages)
...
-              else
-                message = e.message
-              end
-            else
-              message = e.message
-            end
+            message = if [ActiveMerchant::ResponseError, ActiveMerchant::Shipping::ResponseError].include?(e.class) && e.response.is_a?(ActiveMerchant::Shipping::Response)
+              errors = e.response.params.dig('Response', 'Error')
+              errors.map { |error| error.fetch('ErrorDescription') }.join(', ') if errors&.any?
+            end || e.message
```
Comment (mbj) [inline app/models/spree/calculator/shipping/active_shipping/base.rb]: Should we extract this entire hunk into a private utility method to deduplicate with the same code in: https://github.com/MountainRoseHerbs/spree_active_shipping/pull/26/files#diff-0ce2a59535f76b85874ab19f0d1b47aeR40

Comment (dkubb) [inline app/models/spree/calculator/shipping/active_shipping/base.rb]: I don't think there's much point. All this code is horrible and is going to be scrapped as soon as we integrate Encidia. At the moment I am considering `spree_active_shipping` as under "hospice care".

Comment (mbj) [inline app/models/spree/calculator/shipping/active_shipping/base.rb]: k

**Pull request merged**